src/main/java/uk/gov/pay/directdebit/payments/params/PaymentViewSearchParams.java

```diff
@@ -29,15 +39,32 @@
     private String reference;
     private Long amount;
     private String mandateId;
+    private String state;
     private SearchDateParams searchDateParams;
     private PaginationParams paginationParams;
     private Map<String, Object> queryMap;
-    
+
+    private static Map<String, String> externalPaymentToInternalStateQueryMap = new HashMap<>(4);
+
+    static {
+        String andStateCondition = " AND t.state = '%s'";
```

Comment:
I find this `andStateCondition` here not really readable. I would much rather put this in the StringBuilder below, see other comment.
The map would be simply: 

```
externalPaymentToInternalStateQueryMap
     .put(ExternalPaymentState.EXTERNAL_STARTED.getState(), NEW);
etc.
```

Reply:
Great suggestion! :) Will change that, not sure how to handle the case with https://github.com/alphagov/pay-direct-debit-connector/pull/223/files#diff-1f282af1a2bd8e2a97f7d49673214372R60 ?

Reply:
We have clarified this via message.